src/html.rs

```diff
@@ -95,88 +93,117 @@ pub fn walk_and_embed_assets(
                         }
                     }
 
+                    enum LinkType {
+                        Icon,
+                        Stylesheet,
+                        Preload,
+                        DnsPrefetch,
+                        Unknown,
+                    }
+
+                    let mut link_type = LinkType::Unknown;
                     for attr in attrs_mut.iter_mut() {
                         if &attr.name.local == "rel" {
-                            if is_icon(attr.value.as_ref()) {
-                                link_type = "icon";
+                            let value = attr.value.as_ref();
+                            if is_icon(value) {
+                                link_type = LinkType::Icon;
                                 break;
-                            } else if attr.value.as_ref() == "stylesheet" {
-                                link_type = "stylesheet";
+                            } else if value == "stylesheet" {
```

Comment:
I'm wondering, what if those values have a different case? I believe there was a similar issue with icon types, where we have to compare those strings ignoring the case, since `"SHORTCUT ICON"` works exactly the same in browsers, and I found a couple of websites which have a horror like that in their HTML code.

Reply:
I didn't consider the point. Thank you. I think using `eq_ignore_ascii_case` is better. I'll update this branch.

Reply:
Yeah, I think that's what can be found in other parts of the codebase. I'm also not sure if browsers trim attribute values or not. If they do we may be better off moving that comparison out as a global function during code refactor, in order to make monolith behave as close as possible to how browsers would parse the page.

Reply:
I addressed this at 6f158dc.

Reply:
Let me try in my local.

Reply:
I tried below HTML:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>test</title>
    <link rel=" preload " href="should be removed" as="script">
  </head>
  <body>
  </body>
</html>
```

and Chrome did not trim property value as follows:

![スクリーンショット 2020-01-10 14 08 14](https://user-images.githubusercontent.com/823277/72127379-c9240e00-33b2-11ea-85a4-a6826fd481bf.png)

but Chrome tried to fetch the `should be removed` resource. This behavior may be depending on browser's implementation, but we should consider it anyway.

I'll update this branch. Thank you for the point.

Reply:
I trimmed attribute values at b51f41f.

Reply:
Thank you for investigating it and making this feature even more thorough!